metabase/metabase
# Allow editing the breakout bin from the compare aggregation picker and provide presets

Closes https://github.com/metabase/metabase/issues/46531

### Description

This PR:
- allows users to edit the breakout temporal binning directly from the aggregation picker,
- adds presets for offsets
- fixes some small ui inconsistencies

### How to verify
1. New -> Question -> Orders
2. Add aggregation -> Count of rows
3. Add aggregation -> Compare to the past
4. Play around with the popover to configure your aggregations
  a. Add an offset comparison by from a preset:
    1. Click "Previous year"
    2. Click "Done"
    3. Verify that a breakout for "Created At: Year" was added
  b. Add an offset comparison by from a preset:
    1. Click "Previous year"
    2. Customize the offset and unit (eg. "2 Years ago")
    3. Click "Done"
    4. Verify that a breakout for "Created At: Year" was added
  c. Add a comparison with a moving average
    1. Click "Moving average"
    2. Customize the offset and unit (eg. "2 Year moving average")
    3. Click "Done"
    4. Verify that a breakout for "Created At: Year" was added
  d. (the same as c. but with "include this year" selected)
6. Verify that the expression for the newly added aggregations make sense, as well as their names


### Demo

<img width="503" alt="Screenshot 2024-08-15 at 18 07 32" src="https://github.com/user-attachments/assets/d13b0477-12a5-4259-9cbf-6929d7cef34d">
<img width="499" alt="Screenshot 2024-08-15 at 18 07 38" src="https://github.com/user-attachments/assets/acb8fa48-ad1d-468f-9f83-022d23d420f2">

https://github.com/user-attachments/assets/c75f44d2-ba7b-4497-aafd-c4b4067e12a6
